### PR TITLE
Feature #143 GPGS for Profile

### PR DESCRIPTION
**Pull request by EugeneButusov**

### CODE REVIEW

File: Soomla/domain/CCLeaderboard.h [hunk 1]
```diff
@@ -14,8 +25,10 @@
 
 namespace soomla {
     class CCLeaderboard : public CCDomain {
```
Comment (borisspektor) [inline Soomla/domain/CCLeaderboard.h:27]: @EugeneButusov should this still be CCDomain?

Comment (EugeneButusov) [inline Soomla/domain/CCLeaderboard.h:27]: @borisspektor, to unify coding approach (CCUserProfile has such inheritance model) and to use this class https://github.com/soomla/soomla-cocos2dx-core/blob/master/Soomla/CCDomainFactory.h for easy serialization/deserialization.

File: Soomla/CCProfileConsts.cpp [hunk 2]
```diff
@@ -89,17 +91,19 @@ namespace soomla {
     char const *CCProfileConsts::EVENT_INVITE_FAILED = "com.soomla.profile.events.social.InviteFailedEvent";
     char const *CCProfileConsts::EVENT_INVITE_CANCELLED = "com.soomla.profile.events.social.InviteFailedEvent";
 
-    char const *CCProfileConsts::EVENT_UP_GET_LEADERBOARDS_STARTED = "com.soomla.profile.events.social.GetLeaderboardsStartedEvent";
-    char const *CCProfileConsts::EVENT_UP_GET_LEADERBOARDS_FINISHED = "com.soomla.profile.events.social.GetLeaderboardsFinishedEvent";
-    char const *CCProfileConsts::EVENT_UP_GET_LEADERBOARDS_FAILED = "com.soomla.profile.events.social.GetLeaderboardsFailedEvent";
+    char const *CCProfileConsts::EVENT_GET_LEADERBOARDS_STARTED = "com.soomla.profile.events.social.GetLeaderboardsStartedEvent";
+    char const *CCProfileConsts::EVENT_GET_LEADERBOARDS_FINISHED = "com.soomla.profile.events.social.GetLeaderboardsFinishedEvent";
+    char const *CCProfileConsts::EVENT_GET_LEADERBOARDS_FAILED = "com.soomla.profile.events.social.GetLeaderboardsFailedEvent";
```
Comment (vedi) [inline Soomla/CCProfileConsts.cpp:97]: All profile constants use prefix "UP", I'm not sure you should remove it.

Comment (EugeneButusov) [inline Soomla/CCProfileConsts.cpp:97]: @vedi, all literals above doesn't have this prefix. I did it to unify coding style.

**Pull request merged**